pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -383,16 +393,28 @@ def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Union[ReduceOp,
         return tensor
 
     def training_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        if isinstance(self.model, DistributedDataParallel):
+            return self.model(*args, **kwargs)
+        else:
+            return self.model.training_step(*args, **kwargs)
 
     def validation_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        if isinstance(self.model, DistributedDataParallel):
+            return self.model(*args, **kwargs)
+        else:
+            return self.model.validation_step(*args, **kwargs)
 
     def test_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        if isinstance(self.model, DistributedDataParallel):
+            return self.model(*args, **kwargs)
+        else:
+            return self.model.test_step(*args, **kwargs)
```

Comment:
```suggestion
        return self.lightning_module.test_step(*args, **kwargs)
```
we shouldn't wrap with distributed data parallel in testing

Reply:
I'm not sure I understand it. Why couldn't we warp with ddp in testing?